matlab_formatter/matlab_formatter.py

```diff
@@ -401,41 +402,42 @@ def formatFile(self, filename, start, end):
 
 
 def main():
-    options = dict(startLine=1, endLine=None, indentWidth=4,
-                   separateBlocks=True, indentMode='all_functions')
-    indentModes = dict(all_functions=1, only_nested_functions=-1, classic=0)
-
-    if len(sys.argv) < 2:
-        usage = 'usage: matlab_formatter.py filename [options...]\n'
-        opt = '  OPTIONS:\n'
-        for key in options:
-            val = options[key]
-            key_type = re.match(r'\<class \'(.*)\'\>', str(type(val))).group(1)
-            key_type = key_type.replace('NoneType', 'int')
-            opt += '    --%s=%s\n' % (key, key_type)
-
-        print('%s%s' % (usage, opt), file=sys.stderr)
-
-    else:
-        for arg in sys.argv[2:]:
-            key, value = arg.split('=')
-            if any(char.isdigit() for char in value):
-                value = int(value)
-            elif value.lower() == 'true':
-                value = True
-            elif value.lower() == 'false':
-                value = False
-            options[key.strip()] = value
-
-        indent = options['--indentWidth']
-        start = options['--startLine']
-        end = options['--endLine']
-        sep = options['--separateBlocks']
-        mode = indentModes.get(options['--indentMode'],
-                               indentModes['all_functions'])
-
-        formatter = Formatter(indent, sep, mode)
-        formatter.formatFile(sys.argv[1], start, end)
+
+    class indentModes(enum.Enum):
+        all = 1
+        only_nested_functions = -1
+        classic = 0
+
+        def __str__(self):
+            return self.name
+
+        @staticmethod
+        def from_string(s):
+            try:
+                return indentModes[s]
+            except KeyError:
+                raise ValueError()
+
+    ap = argparse.ArgumentParser("matlab formatter main script", formatter_class=argparse.RawTextHelpFormatter)
+    ap.add_argument("filename", type=str, help="Input file name")
+    ap.add_argument("--startLine", type=int, default=1, metavar='', help="")
+    ap.add_argument("--endLine", type=int, metavar='', help="")
+    ap.add_argument("--indentWidth", type=int, default=4, metavar='', help="number of spaces used for indentation")
+    ap.add_argument("--separateBlocks", type=bool, default=True, help="add newlines before and after blocks (for, if, etc.)")
```

Comment:
This line breaks the semantics.  The character string 'false' is translated to true.

Reply:
Fixed and pushed. On the python side I redef'd the `--separateBlocks` arg as a boolean flag. On the js side, I pass in the flag or an nothing per the boolean value from the UI checkbox.